pyrdp/logging/formatters.py

```diff
@@ -7,30 +7,62 @@
 """
 Contains custom logging handlers for the library.
 """
+
 import binascii
 import json
 import logging
 from datetime import datetime
 
 
+class VariableFormatter(logging.Formatter):
+    """
+    Formatter class that provides custom format variables with default values.
+    """
+
+    def __init__(self, fmt: str = None, datefmt: str = None, style: str = "%", defaultVariables: dict = None):
+        super().__init__(fmt = fmt, datefmt = datefmt, style = style)
+        self.defaultVariables = defaultVariables if defaultVariables is not None else {}
```

Comment:
When you use an object as a default value, python uses the same object for every call instead of creating a new object every time. You have to use a workaround like this.